gammapy/spectrum/make.py

```diff
@@ -0,0 +1,220 @@
+import logging
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.utils import lazyproperty
+from regions import CircleSkyRegion
+from gammapy.irf import EnergyDependentMultiGaussPSF, apply_containment_fraction
+from gammapy.maps import WcsGeom
+from gammapy.maps.geom import frame_to_coordsys
+from .core import CountsSpectrum
+from .dataset import SpectrumDataset
+
+log = logging.getLogger(__name__)
+
+
+class SpectrumDatasetMaker:
+    """Make spectrum for a single IACT observation.
+
+    The irfs and background are computed at a single fixed offset,
+    which is recommend only for point-sources.
+
+    Parameters
+    ----------
+    region : `~regions.SkyRegion`
+        Region to compute spectrum dataset for.
+    e_reco : `~astropy.units.Quantity`
+        Reconstructed energy binning
+    e_true : `~astropy.units.Quantity`
+        True energy binning
+    containment_correction : bool
+        Apply containment correction for point sources and circular on regions.
+    binsz : `~astropy.coordinates.Angle`
+        Reference map bin size.
+    width : `~astropy.coordinates.Angle`
+        Reference map width, should encompass the whole region.
+
+    """
+
+    def __init__(
+        self,
+        region,
+        e_reco,
+        e_true=None,
+        containment_correction=True,
+        binsz="0.01 deg",
+        width="0.5 deg",
+    ):
+        self.region = region
+        self.e_reco = e_reco
+        self.e_true = e_true or e_reco
+        self.containment_correction = containment_correction
+        self.binsz = Angle(binsz)
+        self.width = Angle(width)
+
+    # TODO: move this to a RegionGeom class
+    @lazyproperty
+    def geom_ref(self):
+        """Reference geometry to project region"""
+        coordsys = frame_to_coordsys(self.region.center.frame.name)
+        return WcsGeom.create(
+            skydir=self.region.center,
+            width=self.width,
+            binsz=self.binsz,
+            proj="TAN",
+            coordsys=coordsys,
+        )
+
+    @lazyproperty
+    # TODO: move this to a RegionGeom class
+    def region_solid_angle(self):
```

Comment:
these functions are more helper functions for regions than members of the `SpectrumDatasetMaker`. In particular they might be used for the `ReflectedRegionsBackgroundMaker` as well.